src/deluge/dsp/filter/filter_set.h

```diff
@@ -18,119 +18,85 @@
-	void renderHPFLong(q31_t* outputSample, q31_t* endSample, FilterSetConfig* filterSetConfig, int32_t numSamples,
-	                   int32_t sampleIncrement = 1);
-	void renderLadderHPF(q31_t* outputSample, HPLadderConfig* filterSetConfig, int32_t extraSaturation = 0);
 	void reset();
-	BasicFilterComponent lpfLPF1;
-	BasicFilterComponent lpfLPF2;
-	BasicFilterComponent lpfLPF3;
-	BasicFilterComponent lpfLPF4;
+	q31_t set_config(q31_t lpfFrequency, q31_t lpfResonance, bool doLPF, q31_t hpfFrequency, q31_t hpfResonance,
+	                 bool doHPF, LPFMode lpfType, q31_t filterGain, bool adjustVolumeForHPFResonance = true,
+	                 q31_t* overallOscAmplitude = NULL);
+	void copy_config(FilterSet*);
 
-	SVFilter svf;
-
-	BasicFilterComponent hpfHPF1;
-	BasicFilterComponent hpfLPF1;
-	BasicFilterComponent hpfHPF3;
-	uint32_t hpfLastWorkingValue;
-	bool hpfDoingAntialiasingNow;
-	int32_t hpfDivideByTotalMoveabilityLastTime;
-	int32_t hpfDivideByProcessedResonanceLastTime;
-
-	bool hpfOnLastTime;
-	bool lpfOnLastTime;
-	inline void renderLPLadder(q31_t* startSample, q31_t* endSample, LPLadderConfig* filterSetConfig, LPFMode lpfMode,
-	                           int32_t sampleIncrement, int32_t extraSaturation, int32_t extraSaturationDrive);
-	inline void renderLPSVF(q31_t* startSample, q31_t* endSample, LPSVFConfig* filterSetConfig,
-	                        int32_t sampleIncrement);
-	inline void renderLong(q31_t* outputSample, q31_t* endSample, FilterSetConfig* filterSetConfig, LPFMode lpfMode,
-	                       int32_t numSamples, int32_t sampleIncrememt = 1) {
+	inline void renderLong(q31_t* startSample, q31_t* endSample, int32_t numSamples, int32_t sampleIncrememt = 1,
+	                       int32_t extraSaturation = 1) {
 
 		// Do HPF, if it's on
-		if (filterSetConfig->doHPF) {
-			renderHPFLong(outputSample, endSample, filterSetConfig, numSamples, sampleIncrememt);
+		if (HPFOn) {
+			renderHPFLong(startSample, endSample, lpfMode, sampleIncrememt);
 		}
-		else
+		else {
 			hpfOnLastTime = false;
-
+		}
+		// Do LPF, if it's on
+		if (LPFOn) {
+			renderLPFLong(startSample, endSample, lpfMode, sampleIncrememt, extraSaturation, extraSaturation >> 1);
+		}
+		else {
+			lastLPFMode = LPFMode::OFF;
+		}
+	}
+	inline void renderLongStereo(q31_t* startSample, q31_t* endSample, int32_t extraSaturation = 1) {
+		// Do HPF, if it's on
+		if (HPFOn) {
+			renderHPFLongStereo(startSample, endSample, extraSaturation);
+		}
+		else {
+			hpfOnLastTime = false;
+		}
 		// Do LPF, if it's on
-		if (filterSetConfig->doLPF) {
-			renderLPFLong(outputSample, endSample, filterSetConfig, lpfMode, sampleIncrememt, 1, 1);
+		if (LPFOn) {
+			renderLPFLongStereo(startSample, endSample, extraSaturation);
+		}
+		else {
+			lastLPFMode = LPFMode::OFF;
 		}
-		else
-			lpfOnLastTime = false;
 	}
 
+	//used to check whether the filter is used at all
```

Comment:
Since these are only used separately in the copy routine (which I think could just directly read the private members) and by the higher-level voice rendering code (which only cares if any filter is engaged) perhaps the public API here should just be `isAnyFilterOn()`

Reply:
I considered it but thought this might be good for future expansion